Assets/Scripts/Map/ScreenUIController.cs

```diff
@@ -0,0 +1,67 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace Assets.Scripts.Map
+{
+    public class ScreenUIController : MonoBehaviour
+    {
+        [SerializeField] private GameObject _menuUi;
+        [SerializeField] private GameObject _optionsUi;
+        [SerializeField] private GameObject _levelUpUi;
+        [SerializeField] private GameObject _winScreenUi;
+        [SerializeField] private GameObject _loseScreenUi;
+        private Map _map;
+
+        public void AwakeController()
+        {
+            var mapController = CreateMapController();
+            _map = new Map(mapController);
+        }
+        public void Menu() { _map.OpenMenu(); }
+        public void Resume() { _map.Resume(); }
+        public void Options() { _map.OpenOptions(); }
+        public void LevelUp() { _map.InitiateLevelUp(); }
+        public void Win() { _map.Win(); }
+        public void Lose() { _map.Lose(); }
+
+        private IMapController CreateMapController()
+        {
+            if (!IsUIObjectsNotNull()) return null;
```

Comment:
Кажется, что проверка на NULL здесь лишняя - к названию CreateMapController не вполне относится.
Лучше перенести в конструктор. Кроме того, возвращать null в качестве IMapController не хорошо, код Map и далее, всех классов-состояний не проверяет на null (да и не должен) и будет падать. 
Думаю, лучше возбуждать исключения (здесь весьма уместно). Эти исключения - для тебя (а не для конечного пользователя), чтобы ты узнал, что почему-то Unity не проинициализировал [SerializedField]s.
```suggestion
            private void CheckSerializedFields()
            {
                if (_menuUi == null) throw ArgumentNullException(nameof(_menuUi));
                //...
                if (_loseScreenUi == null) throw ArgumentNullException(nameof(_loseScreenUi));
            }
```
Конечно, в этом случае ArgumentNullException используется чуть не по назначению, т.к. проверяются поля, а не аргументы, но думаю, тут это уместно.

Как вариант - перенести этот код в конструктор SimpleMapController, там проверка будет лучше всего - она и аргументы будет проверять, а не поля, и проверка будет ближе к использованию, это хорошо, т.к. труднее забыть проверить какое-нибудь новое поле.

Reply:
Перенёс проверку на Null в конструктор SimpleMapController